rtshell/rtfsm.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python2
+# -*- Python -*-
+# -*- coding: utf-8 -*-
+
+'''rtshell
+
+Copyright (C) 2009-2015
+    Geoffrey Biggs
+    RT-Synthesis Research Group
+    Intelligent Systems Research Institute,
+    National Institute of Advanced Industrial Science and Technology (AIST),
+    Japan
+    All rights reserved.
+Licensed under the GNU Lesser General Public License version 3.
+http://www.gnu.org/licenses/lgpl-3.0.en.html
+
+Implementation of the fsm configuration command.
+
+'''
+
+from __future__ import print_function
+
+
+import optparse
+import os
+import rtctree.exceptions
+import rtctree.path
+import rtctree.tree
+import rtctree.utils
+import rtctree.rtc.RTC
+import sys
+import traceback
+
+from rtshell import path
+from rtshell import rts_exceptions
+import rtshell
+
+
+def manage_fsm(tgt_raw_path, tgt_full_path, command, argument, options, tree=None):
+    path, port = rtctree.path.parse_path(tgt_full_path)
+    if port:
+        raise rts_exceptions.NotAComponentError(tgt_raw_path)
+    if not path[-1]:
+        raise rts_exceptions.NotAComponentError(tgt_raw_path)
+
+    if not tree:
+        tree = rtctree.tree.RTCTree(paths=path, filter=[path])
+
+    if not tree.has_path(path):
+        raise rts_exceptions.NoSuchObjectError(path)
+    rtc = tree.get_node(path)
+    if rtc.is_zombie:
+        raise rts_exceptions.ZombieObjectError(path)
+    if not rtc.is_component:
+        raise rts_exceptions.NotAComponentError(path)
+
+    fsm = rtc.get_extended_fsm_service()
+
+    if command == 'getstate':
+        print(fsm.get_current_state())
+    elif command == 'geteventprofiles':
+        (ret, struct) = fsm.get_fsm_structure()
+        ps = []
+        for p in struct.event_profiles:
+            ps.append(p.name + ':' + p.data_type)
+        print(','.join(ps))
```

Comment:
Return a list of strings instead of printing.

There should be a space after the comma.